src/Nethermind/Nethermind.Synchronization/FastBlocks/FastHeadersSyncFeed.cs

```diff
@@ -180,19 +181,20 @@ protected virtual void PostFinishCleanUp()
             _syncReport.HeadersInQueue.MarkEnd();
         }
 
-        private void HandleDependentBatches()
+        private void HandleDependentBatches(CancellationToken cancellationToken)
         {
             long? lowest = LowestInsertedBlockHeader?.Number;
             while (lowest.HasValue && _dependencies.TryRemove(lowest.Value - 1, out HeadersSyncBatch? dependentBatch))
             {
                 InsertHeaders(dependentBatch!);
                 lowest = LowestInsertedBlockHeader?.Number;
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Do we really want to throw instead of just checking if cancellation was requested?

Reply:
Where is this exception caught?

Reply:
In SyncDispatcher, changes Feed to Finished state